raiden/blockchain/abi.py

```diff
@@ -69,6 +77,18 @@ def get_eventname_types(event_description):
     return name, encode_types
 
 
+#  Initially, I wanted to use solc's AST output, but that thing is so crazy that parsing
+#    it would require writing a new module to get just one solidity variable.
+#  So I opted for good ol' regex..
+def parse_contract_version(contract_file):
+    contract_file = get_contract_path(contract_file)
+    original = open(contract_file, 'r')
```

Comment:
Sure, I'll correct that.
On the bottom line I've never understood appeal of context managers. To me it is superfluous to add one more level of indent where an exception is raised anyway in case of error and GC will do its job cleaning up the resource eventually.

Reply:
As far as I understand the context manager takes care of closing the OS-level file descriptors, which aren't handled by the GC and will leak otherwise.

Reply:
I don't think that's the case. Try this.
```
import time
import os

def open_me():
    f = open("/etc/passwd", "r")
    f.readlines()

for i in range(1,100):
    open_me()

print os.listdir("/proc/%d/fd" % os.getpid())
```
```
➜  /tmp py x.py
['0', '1', '2', '3']
```
(0,1,2 are std(in|out|err), 3 is fd opened by `listdir`)
Descriptor is properly closed by `__del__` method of the `file` object. I can imagine with some high-performance app you can eventually run out of fds though if you rely solely on GC to do cleanup...

Reply:
There is really no reason to avoid using deterministic resource management strategy, specially if the argument is one indentation level. Syntax is not more important than semantics.

Yes, the code above will not cause Raiden to run out of fds. Even if a reference is leaked because of an exception, the process will exit and there are no resources to manage at this point. Nevertheless it's good code style and goes along with the rest of the codebase.

Reply:
Interesting, I thought it wouldn't clean up that properly.